plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -72,15 +87,66 @@ rules :: Recorder (WithPriority Log) -> PluginId -> Rules ()
 rules recorder plId = do
   define (cmapWithPrio LogShake recorder) $
     \GetStanDiagnostics file -> do
-      config <- getPluginConfigAction plId
-      if pluginEnabledConfig plcDiagnosticsOn config then do
+      plugConfig <- getPluginConfigAction plId
+      if pluginEnabledConfig plcDiagnosticsOn plugConfig then do
           maybeHie <- getHieFile file
           case maybeHie of
             Nothing -> return ([], Nothing)
             Just hie -> do
-              let enabledInspections = HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)]
-              -- This should use Cabal config for extensions and Stan config for inspection preferences is the future
-              let analysis = runAnalysis Map.empty enabledInspections [] [hie]
+              let currentHSfromHIEAbs = hie_hs_file hie
+              currentHSfromHIERel <- liftIO $ makeRelativeToCurrentDirectory currentHSfromHIEAbs
+              -- This codes follows what 'runStan' does, from the module 'Stan'
+
+              -- There aren't any cli args. isLoud=False=Silent output
+              let isLoud = False -- Should this be enabled when debugging? Enables default stan cli output
+              let stanArgsConfigFile = Nothing -- There aren't any cli args
+
+              EnvVars{envVarsUseDefaultConfigFile} <- liftIO getEnvVars
+              logWith recorder Debug (LogDebug $
+                "envVarsUseDefaultConfigFile: " <> (T.pack $ show envVarsUseDefaultConfigFile))
+
+              let defConfTrial = envVarsUseDefaultConfigFile -- There aren't any cli args: <> stanArgsUseDefaultConfigFile
+              let useDefConfig = maybe True snd (trialToMaybe defConfTrial)
+
+              tomlConfig <- liftIO $ getTomlConfig isLoud useDefConfig stanArgsConfigFile
+              let configTrial = finaliseConfig $ defaultConfig <> tomlConfig -- There aren't any cli args: <> stanArgsConfig
+              logWith recorder Debug (LogDebug $ "Final stan config result\n" <> ( T.pack $ show configTrial))
+
+              (cabalExtensionsMap, checksMap, confIgnored) <- case configTrial of
+                   FiascoL es -> do
+                      logWith recorder Warning (LogWarnConf es)
+                      pure (Map.empty,
+                            HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)],
+                            [])
+                   ResultL warnings stanConfig -> do
+                      -- I'm not sure this is the best way to obtain the .cabal
+                      -- for this file but it'll have to do. Anyways, if it is not
+                      -- found it's not a big issue. That was the default previously.
+                      maybeCabalFileDir <- let maybeCabalFileDir = findFileUpwardsF
```

Comment:
Also I don't know if you looked around but I bet we do this somewhere else in HLS also...